src/operator/controllers/intents_reconcilers/kafka_acls.go

```diff
@@ -124,36 +124,28 @@ func (r *KafkaACLReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	// examine DeletionTimestamp to determine if object is under deletion
-	if intents.ObjectMeta.DeletionTimestamp.IsZero() {
-		if !controllerutil.ContainsFinalizer(intents, KafkaACLsFinalizerName) && intents.HasKafkaTypeInCallList() {
-			logger.Infof("Adding finalizer %s", KafkaACLsFinalizerName)
-			controllerutil.AddFinalizer(intents, KafkaACLsFinalizerName)
-			if err := r.client.Update(ctx, intents); err != nil {
-				return ctrl.Result{}, err
-			}
-		}
-	} else {
-		// The object is being deleted
-		if controllerutil.ContainsFinalizer(intents, KafkaACLsFinalizerName) {
-			logger.Infof("Removing associated Acls")
-			if err := r.RemoveACLs(intents); err != nil {
-				r.RecordWarningEventf(intents, ReasonRemovingKafkaACLsFailed, "Could not remove Kafka ACLs: %s", err.Error())
-				return ctrl.Result{}, err
-			}
+	if r.intentsObjectUnderDeletion(intents) {
+		return r.handleIntentsDeletion(ctx, intents, logger)
+	}
 
-			controllerutil.RemoveFinalizer(intents, KafkaACLsFinalizerName)
-			if err := r.client.Update(ctx, intents); err != nil {
-				if k8serrors.IsConflict(err) {
-					return ctrl.Result{Requeue: true}, nil
-				}
-				return ctrl.Result{}, err
-			}
+	if r.isMissingKafkaFinalizer(intents) {
+		logger.Infof("Adding finalizer %s", KafkaACLsFinalizerName)
+		controllerutil.AddFinalizer(intents, KafkaACLsFinalizerName)
+		if err := r.client.Update(ctx, intents); err != nil {
+			return ctrl.Result{}, err
 		}
+	}
 
-		return ctrl.Result{}, nil
+	result, err2 := r.applyAcls(logger, err, intents)
```

Comment:
Use `err` instead of `err2`
GoLand likes to suggest err2 when refactoring, but it's uncommon in actual code - I suggest re-using `err`.